src/helpers/APIError.js

```diff
@@ -1,35 +1,115 @@
-import httpStatus from 'http-status';
+import { initLogLevelGte } from 'winston-json-formatter';
+import config from '../config/config';
+
+// Read https://www.joyent.com/node-js/production/design/errors to understand the difference between
+// "programmer errors" (i.e. bugs) and "operational errors" (i.e. not actual bugs).
+// The APIError class assumes that if it was instantiated without a causal error, then it is being
+// used to define an "operational error", and there is no actual "programmer error"/bug.
```

Comment:
Mmm. Per our discussion I actually had it the other way at first, but I kept getting confused because the more verbose levels correspond to higher integers.